modules/low-watermark/src/main/java/org/apache/ignite/internal/lowwatermark/LowWatermarkImpl.java

```diff
@@ -234,18 +234,12 @@ HybridTimestamp createNewLowWatermarkCandidate() {
     }
 
     private void setLowWatermark(HybridTimestamp newLowWatermark) {
-        updateLowWatermarkLock.writeLock().lock();
```

Comment:
Why is protection with the write lock removed?

Reply:
This method is called only from one place `waitForLocksAndSetLowWatermark`, where the corresponding write lock is already acquired.